pkg/minishift/openshift/service.go

```diff
@@ -214,15 +218,29 @@ func filterAndUpdateServiceSpecs(routeSpec map[string][]ServiceWeight, namespace
 				delete(serviceNodePort, service.Name)
 			} else {
 				if _, ok := serviceSpecMap[service.Name]; !ok {
-					serviceSpecMap[service.Name] = &ServiceSpec{Namespace: namespace,
-						Name:     service.Name,
-						URL:      []string{fmt.Sprintf("http://%s", routeURL)},
-						NodePort: "",
-						Weight:   []string{service.Weight},
+					if service.Tls != "" {
+						serviceSpecMap[service.Name] = &ServiceSpec{Namespace: namespace,
+							Name:     service.Name,
+							URL:      []string{fmt.Sprintf("https://%s", routeURL)},
+							NodePort: "",
+							Weight:   []string{service.Weight},
+						}
+					} else {
+						serviceSpecMap[service.Name] = &ServiceSpec{Namespace: namespace,
+							Name:     service.Name,
+							URL:      []string{fmt.Sprintf("http://%s", routeURL)},
+							NodePort: "",
+							Weight:   []string{service.Weight},
+						}
 					}
 				} else {
-					serviceSpecMap[service.Name].URL = append(serviceSpecMap[service.Name].URL, fmt.Sprintf("http://%s", routeURL))
-					serviceSpecMap[service.Name].Weight = append(serviceSpecMap[service.Name].Weight, service.Weight)
+					if service.Tls != "" {
+						serviceSpecMap[service.Name].URL = append(serviceSpecMap[service.Name].URL, fmt.Sprintf("https://%s", routeURL))
+						serviceSpecMap[service.Name].Weight = append(serviceSpecMap[service.Name].Weight, service.Weight)
+					} else {
+						serviceSpecMap[service.Name].URL = append(serviceSpecMap[service.Name].URL, fmt.Sprintf("http://%s", routeURL))
+						serviceSpecMap[service.Name].Weight = append(serviceSpecMap[service.Name].Weight, service.Weight)
+					}
```

Comment:
you need to add a unit test also for this change, check https://github.com/minishift/minishift/blob/master/pkg/minishift/openshift/service_test.go we have a dummy testdata for route and service spec you can add another data which have tls and then check if it returns `https` uri for route.

Reply:
Okay

Reply:
@praveenkumar Added the unit test, PTAL